php/meterpreter/meterpreter.php

```diff
@@ -35,6 +35,13 @@
   $GLOBALS['id2f'] = array();
 }
 
+if (!defined('AF_INET')) {
+    define('AF_INET', 2);
+}
+if (!defined('AF_INET6')) {
+    define('AF_INET6', 23);
+}
```

Comment:
For me on my mac AF_INET6 is 30:
```
php -a
Interactive shell

php > echo AF_INET;
2
php > echo AF_INET6;
30
```

Other constants:
```
php > $a = get_defined_constants(TRUE) ; foreach ( $a['sockets'] as $constant => $value ) { printf("%-25s %d\r\n", $constant, $value) ; }
AF_UNIX                   1
AF_INET                   2
AF_INET6                  30
```

And on my kali box:
```
php -a
Interactive mode enabled

php > echo AF_INET;
2
php > echo AF_INET6;
10
php > 
```

Reply:
Looks like the constants should align with meterpreter's view of the world anyways:

https://github.com/rapid7/metasploit-framework/blob/3e63fe579ff7e97bdb75069b3e120166cfb101d4/lib/rex/post/meterpreter/extensions/stdapi/constants.rb#L14-L20

There's probably cross-platform bugs here for the other meterpreters; But I guess we should define our own constants instead of using AF_INET/AF_INET6 to stop conflicts here

Reply:
This change only defines them if they are undefined, in which case it'll prevent a crash. It's not perfect but certainly an improvement.

Reply:
Sorry, to clarify - we shouldn't use php's definition of AF_INET/AF_NET6 here at all - as those values will change depending on the host. We should introduce new constants values that align with what msfconsole has defined in the Ruby codebase, which I believe were taken from Window's definition of AF_INET6 if I understand correctly https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-socket

Reply:
Oops I understand now. Let me send a pull request which hopefully resolves this.

Reply:
Please see https://github.com/rapid7/metasploit-payloads/pull/562